lib/galaxy/datatypes/binary.py

```diff
@@ -1017,30 +1017,30 @@ def set_meta(self, dataset, overwrite=True, **kwd):
             f = h5py.File(dataset.file_name)
             attributes = dict(f.attrs.items())
 
-            dataset.metadata.id = attributes['id']
-            dataset.metadata.format_url = attributes['format-url']
+            dataset.metadata.id = util.unicodify(attributes['id'])
+            dataset.metadata.format_url = util.unicodify(attributes['format-url'])
             if 'format-version' in attributes:  # biom 2.1
-                dataset.metadata.format_version = '.'.join(map(str, list(attributes['format-version'])))
+                dataset.metadata.format_version = '.'.join((util.unicodify(_) for _ in list(attributes['format-version'])))
             elif 'format' in attributes:  # biom 2.0
-                dataset.metadata.format = attributes['format']
-            dataset.metadata.type = attributes['type']
-            dataset.metadata.shape = tuple(attributes['shape'])
-            dataset.metadata.generated_by = attributes['generated-by']
-            dataset.metadata.creation_date = attributes['creation-date']
+                dataset.metadata.format = util.unicodify(attributes['format'])
+            dataset.metadata.type = util.unicodify(attributes['type'])
+            dataset.metadata.shape = tuple((util.unicodify(_) for _ in attributes['shape']))
```

Comment:
These are numbers in a `numpy.ndarray`, so the change is not necessary.
```suggestion
            dataset.metadata.shape = tuple(attributes['shape'])
```

Reply:
I thought so too, but https://jenkins.galaxyproject.org/job/docker-integration-py3/202/testReport/test.integration/test_datatype_upload/test_upload_datatype_auto_biom2_sparse_otu_table_hdf5_biom2_/ led me to add the https://github.com/galaxyproject/galaxy/pull/8497/commits/3dd500478df5e0db96de502fe14569cf92462d75

Reply:
ah, I guess they are some of the numpy numbers, like np.int8 or something like that, I can cast to int instead (if these are always int).